test/bigtable/table.js

```diff
@@ -0,0 +1,852 @@
+
+      table.deleteRows(assert.ifError);
+    });
+
+    it('should respect the row key prefix option', function(done) {
+      var options = {
+        prefix: 'a'
+      };
+
+      table.request = function(g, reqOpts) {
+        assert.strictEqual(reqOpts.rowKeyPrefix, options.prefix);
+
+        var spy = FakeMutation.convertToBytes;
+
+        assert.strictEqual(spy.callCount, 1);
+        assert.strictEqual(spy.getCall(0).args[0], options.prefix);
+        done();
+      };
+
+      table.deleteRows(options, assert.ifError);
+    });
+
+    it('should delete all data when no options are provided', function(done) {
+      table.request = function(g, reqOpts) {
+        assert.strictEqual(reqOpts.deleteAllDataFromTable, true);
+        done();
+      };
+
+      table.deleteRows(assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var error = new Error('err');
+      var response = {};
+
+      table.request = function(g, r, callback) {
+        callback(error, response);
+      };
+
+      table.deleteRows(function(err, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('family', function() {
+    var FAMILY_ID = 'test-family';
+
+    it('should create a family with the proper arguments', function() {
+      var family = table.family(FAMILY_ID);
+
+      assert(family instanceof FakeFamily);
+      assert.strictEqual(family.calledWith_[0], table);
+      assert.strictEqual(family.calledWith_[1], FAMILY_ID);
+    });
+  });
+
+  describe('getFamilies', function() {
+    it('should return an error to the callback', function(done) {
+      var error = new Error('err');
+      var response = {};
+
+      table.getMetadata = function(callback) {
+        callback(error, response);
+      };
+
+      table.getFamilies(function(err, families, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(families, null);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+
+    it('should return an array of Family objects', function(done) {
+      var metadata = {
+        a: 'b'
+      };
+
+      var response = {
+        columnFamilies: {
+          test: metadata
+        }
+      };
+
+      table.getMetadata = function(callback) {
+        callback(null, response);
+      };
+
+      var familyFactory = table.family;
+
+      table.family = function(id) {
+        assert.strictEqual(id, 'test');
+        return familyFactory.apply(table, arguments);
+      };
+
+      table.getFamilies(function(err, families, apiResponse) {
+        assert.ifError(err);
+
+        var family = families[0];
+        assert(family instanceof FakeFamily);
+        assert.strictEqual(family.metadata, metadata);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('getRows', function() {
```

Comment:
I like how this test block was done. Testing streams is tricky, and you made it look so easy 👍